docs/content-flow.md

```diff
@@ -0,0 +1,388 @@
+      },
+      {
+         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
+         "size": 99,
+         "digest": "sha256:de36df38e0b6c0e7f29913c68884a0323207c07cd7c1eba71d5618f525ac2ba6"
+      },
+      {
+         "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
+         "size": 410,
+         "digest": "sha256:602cd484ff92015489f7b9cf9cbd77ac392997374b1cc42937773f5bac1ff43b"
+      }
+   ]
+}
+```
+
+We can see that this is a standard image, with one config, whose hash is `sha256:df57482065789980ee9445b1dd79ab1b7b3d1dc26b6867d94470af969a64c8e6` and 6 layers.
+
+Each of these elements - the index, the manifests, the config file and the layers - is stored
+separately in the registry.
+
+### Content Store
+
+When content is loaded into containerd's content store, it stores them very similarly to how the registry does.
+Each component is stored in a file whose name is the hash of it.
+
+Continuing our redis example, if we do `client.Pull()` or `ctr pull`, we will get the following in our
+content store:
+
+* `sha256:1d0b903e3770c2c3c79961b73a53e963f4fd4b2674c2c4911472e8a054cb5728` - the index
+* `sha256:a5aae2581826d13e906ff5c961d4c2817a9b96c334fd97b072d976990384156a` - the manifest for `linux/amd64`
+* `sha256:df57482065789980ee9445b1dd79ab1b7b3d1dc26b6867d94470af969a64c8e6` - the config
+* `sha256:123275d6e508d282237a22fefa5aef822b719a06496444ea89efa65da523fc4b` - layer 0
+* `sha256:f2edbd6a658e04d559c1bec36d838006bbdcb39d8fb9033ed43d2014ac497774` - layer 1
+* `sha256:66960bede47c1a193710cf8bfa7bf5f50bc46374260923df1db1c423b52153ac` - layer 2
+* `sha256:79dc0b596c9027416a627a6237bd080ac9d87f92b60f1ce145c566632839bce7` - layer 3
+* `sha256:de36df38e0b6c0e7f29913c68884a0323207c07cd7c1eba71d5618f525ac2ba6` - layer 4
+* `sha256:602cd484ff92015489f7b9cf9cbd77ac392997374b1cc42937773f5bac1ff43b` - layer 5
+
+If we look in our content store, we see exactly these (I filtered and sorted to make it easier to read):
+
+```sh
+$ tree /var/lib/containerd/io.containerd.content.v1.content/blobs
+/var/lib/containerd/io.containerd.content.v1.content/blobs
+└── sha256
+    ├── 1d0b903e3770c2c3c79961b73a53e963f4fd4b2674c2c4911472e8a054cb5728
+    ├── a5aae2581826d13e906ff5c961d4c2817a9b96c334fd97b072d976990384156a
+    ├── df57482065789980ee9445b1dd79ab1b7b3d1dc26b6867d94470af969a64c8e6
+    ├── 123275d6e508d282237a22fefa5aef822b719a06496444ea89efa65da523fc4b
+    ├── f2edbd6a658e04d559c1bec36d838006bbdcb39d8fb9033ed43d2014ac497774
+    ├── 66960bede47c1a193710cf8bfa7bf5f50bc46374260923df1db1c423b52153ac
+    ├── 79dc0b596c9027416a627a6237bd080ac9d87f92b60f1ce145c566632839bce7
+    ├── de36df38e0b6c0e7f29913c68884a0323207c07cd7c1eba71d5618f525ac2ba6
+    └── 602cd484ff92015489f7b9cf9cbd77ac392997374b1cc42937773f5bac1ff43b
+```
+
+We can see the same thing if we use the containerd interface. Again, we sorted it for consistent easier viewing.
+
+```sh
+ctr content ls
```

Comment:
\`\`\`console
$ ctr content ls
...
\`\`\`

Reply:
Bad habits die hard. I always put in `sh` for these things, but shouldn't. Fixing.